### ISSUE TITLE
Unable to specify keyboard focus indices in Autocomplete Container

### ISSUE THREAD

**Issue by jpandl19** [zendeskgarden/react-components#111]
## Context
I'm working on a simple implementation of an Autocomplete using Garden's [Autocomplete Container](https://garden.zendesk.com/react-components/autocomplete/)

The use case is heavily influenced by the Garden [example](https://garden.zendesk.com/react-components/autocomplete/#autocomplete), however, I'm abstracting away most of the stateful logic and allowing people to use the component via a render prop pattern like so:

``` jsx
      <AutocompleteAdder
        label={`${label}. Menu visible: ${isMenuVisible}`}
        inputValue={inputValue}
        currentValue={currentValue || ''}
        defaultValue={currentValue || defaultValue}
        addItemKey={ADD_ITEM_KEY}
        addItemLabel={'Add new item'}
        onInputChange={this.handleInputChange}
        onOptionSelect={this.handleOptionSelect}
        onMenuVisibilityChange={this.handleMenuVisibilityChange}
      >
        {({ getItemProps, focusedKey }) => (
          <CheckableItemMenu
            menuItems={filterMenuOptions(options, inputValue)}
            selectedKey={selectedKey}
            getItemProps={getItemProps}
            focusedKey={focusedKey}
          />
        )}
      </AutocompleteAdder>
```

Which winds up looking like this: 
![image](https://user-images.githubusercontent.com/11040668/44063997-d104320c-9f17-11e8-8515-7a0b1e95edaa.png)

Within `AutocompleteAdder`, a menu is rendered like so:

```jsx
<MenuView {...menuProps}>
    <MenuItemContainer>{menuRender({ menuProps, getItemProps, focusedKey })}</MenuItemContainer>
    <Separator />
    <AddItem
      {...getItemProps({
        key: addItemKey,
        focused: focusedKey === addItemKey,
      })}
      data-test-id={AUTOCOMPLETE_ADDER_ADD_ITEM}
    >
      {addItemLabel}
    </AddItem>
  </MenuView>
```

## The Problem
React doesn't seem to render things in the order you'd expect it to. In this case, `AddItem`'s `getItemProps` is called *before* all of the item's in `menuRender`. This might be a problem with using the function as a child/ render prop pattern, but in any case it seems we can't rely solely on the React render order for everything. Since keyboard nav ordering is set via the order that `getItemProps` is called in, this means that `AddItem` is focused before the other item's, even though they are "on top of" `AddItem` within the DOM. 

## Expectations
Since React render ordering is not guaranteed/obvious, I need to be able to specify an ordering for the `focusedKeys` within `_this.indexKeyMap[currentIndex] `. Downshift has an example of this here: https://github.com/paypal/downshift/blob/master/src/downshift.js#L789

## Reality
Keyboard nav is not compatible with the render prop/ function as a child HOC pattern.

## Steps to Reproduce

1. Implement an `AutocompleteContainer` component uses the render prop pattern (copy and pasting my code into the Garden storyguist example should work)
2. Observe that keyboard nav appears broken, as it jumps to the `AddItem` before the other items.

###### Fine Print

* Component: Autocomplete
* Browsers: all
